### DISCUSSION TITLE
How to express an Enum that renders as Int in JSON?

### DISCUSSION THREAD

**Discussion by bfloch** [apple/pkl#258] [Q&A] [answered]
I would love to have the strong type guarantee of an Union, but render as an Int into JSON.

What is good way to achieve this?

**Comment by bioball** [comment, marked as answer]
I think the best thing to do here is to add a type constraint, rather than a union.

```groovy
num: Int(this == 1 || this == 2 || this == 3)
```

One other idea is to use a string literal union, and plus a converter. But this feels like a little bit of unnecessary indirection.

```groovy
num: "1"|"2"|"3"

output {
  renderer = new JsonRenderer {
    converters {
      ["num"] = (it) -> it.toInt()
    }
  }
}
```
